src/core/speed.h

```diff
@@ -0,0 +1,73 @@
+#ifndef CORE_SPEED_H
+#define CORE_SPEED_H
+
+#include "core/time.h"
+
+/**
+ * @file
+ * Scroll speed functions.
+ * Calculates deltas for constant speed/acceleration/deceleration based on
+ * a "number of pixels per unit of time" value.
+ */
+
+/**
+ * Constant for immediate speed change with no acceleration/deceleration
+ */
+#define SPEED_CHANGE_IMMEDIATE 0
+
+/**
+* Speed direction constants
+*/
+typedef enum {
+    SPEED_DIRECTION_NEGATIVE = -1,
+    SPEED_DIRECTION_STOPPED  = 0,
+    SPEED_DIRECTION_POSITIVE = 1
+} speed_direction;
+
+typedef struct {
+    time_millis start_time;
+    time_millis total_time;
+    time_millis last_speed_check;
+    int speed_difference;
+    int desired_speed;
+    int current_speed;
+    int cumulative_delta;
+} speed;
+
+/**
+ * Clears speed structure
+ * @param sp Speed structure to clear
+ */
+void speed_clear(speed *sp);
+
+/**
+ * Changes the speed
+ * @param sp Speed structure to act on
+ * @param new_speed The new speed
+ * @param total_time the time for the acceleration/deceleration, or SPEED_CHANGE_IMMEDIATE
+ */
+void speed_change(speed *sp, int new_speed, time_millis total_time);
```

Comment:
Changes the speed in what way? I think something like `speed_set_target_speed` would be a better name.

Reply:
How about just `speed_set_target`? Otherwise the word speed is repeated...

Reply:
True, that's also fine.